channeldb/kvdb/etcd/readwrite_bucket.go

```diff
@@ -24,9 +24,9 @@ type readWriteBucket struct {
 // newReadWriteBucket creates a new rw bucket with the passed transaction
 // and bucket id.
 func newReadWriteBucket(tx *readWriteTx, key, id []byte) *readWriteBucket {
-	if !bytes.Equal(id, tx.rootBucketID[:]) {
-		// Add the bucket key/value to the lock set.
-		tx.lock(string(key), string(id))
+	if !bytes.Equal(key, tx.rootBucketID[:]) {
+		// Add the bucket key to the exclude set.
+		tx.stm.Exclude(string(key))
```

Comment:
👍

Reply:
Yeah, this commit wasn't part of the PR originally but moved here for convenience.